src/activities/parse-docmap.ts

```diff
@@ -1,11 +1,25 @@
-import { parser } from '@elifesciences/docmap-ts';
+import { VersionedReviewedPreprint, parser } from '@elifesciences/docmap-ts';
 
 export const parseDocMap = async (docMapInput: string): Promise<parser.ManuscriptData> => {
   const manuscriptData = parser.parsePreprintDocMap(docMapInput);
 
-  if (!manuscriptData) {
-    throw Error('Could not parse docmap');
-  }
+  try {
+    // TO-DO: This is a temporary hack, remove/replace once DocMaps are fixed to hold the meca path in the output content
+    const { steps } = JSON.parse(docMapInput);
+
+    const contentPaths: string[] = [];
+    Object.keys(steps).forEach((key) => {
+      // Loop through the actions in a step and the corresponding outputs to extract _tdmPaths
+      // eslint-disable-next-line @typescript-eslint/dot-notation
+      steps[key].actions.forEach((action: any) => action.outputs?.forEach((output: any) => { if (output['_tdmPath']) contentPaths.push(output['_tdmPath']); }));
+    });
 
-  return manuscriptData;
+    if (contentPaths.length === manuscriptData?.versions.length) {
+      manuscriptData.versions = manuscriptData.versions.map<VersionedReviewedPreprint>((version, index) => ({ ...version, preprint: { ...version.preprint, content: contentPaths[index] } }));
```

Comment:
There are so many assumptions in here about the structure and makeup of a docmap, but given we invented the _tdmPath I'm guessing it's not going to matter in practice. On that basis, I think it should work OK.

I'm not suggesting doing this, as it's just not valuable enough, but a potential improvement might have been using a Map/object for `contentPaths` to index tdmPaths with their DOIs or the content URL, then hunting for matching DOI/URL in the versions to replace the content URL in versions. This would have at least made sure we're replacing the correct values in the version structure. We then also wouldn't need to error if the number of versions and TDM paths didn't match (which is allowed in a docmap, we just don't do that). 

If [this issue](https://github.com/elifesciences/data-hub-issues/issues/671) is resolved, we'd also need to update this code to look to inputs as well as outputs for TDM paths

Reply:
That improvement would make sense but I agree it works _for now_ until _tdmPath is replaced 😂